backend/tests/endpoints/test_room.py

```diff
@@ -348,3 +373,152 @@ async def test_get_members_unsuccessful(mock_dataStorage_read):
     response = client.get(url=get_room_members_url)
     assert response.status_code == 424
     assert response.json() == {"detail": "Failure to retrieve room members"}
+
+
+@pytest.mark.parametrize(
+    "url, status_code, json_response",
+    [
+        (
+            leave_room_url,
+            200,
+            {
+                "status": "success",
+                "message": "user removed from room successfully",
+                "data": {
+                    "matched_documents": 1,
+                    "modified_documents": 1,
+                    "member_id": "619baa5c1a5f54782939d386",
+                    "room_id": "23dg67l0eba8adb50ca13a24",
+                },
+            },
+        ),
+        (
+            remove_room_member_url,
+            200,
+            {
+                "status": "success",
+                "message": "user removed from room successfully",
+                "data": {
+                    "matched_documents": 1,
+                    "modified_documents": 1,
+                    "member_id": "61696f5ac4133ddaa309dcfe",
+                    "room_id": "23dg67l0eba8adb50ca13a24",
+                },
+            },
+        ),
+        (
+            f"{fake_member_leave_room_url}",
+            404,
+            {"detail": "user not a member of the room"},
+        ),
+        (
+            f"{fake_member_remove_room_member_url}",
+            404,
+            {"detail": "user not a member of the room"},
+        ),
+        (
+            f"{remove_room_member_url}90",
+            404,
+            {"detail": "admin id specified not a member of the room"},
+        ),
+        (
+            f"{non_admin_remove_room_member_url}",
+            403,
+            {"detail": "must be an admin to remove member"},
+        ),
+    ],
+)
+@pytest.mark.asyncio
+@mock.patch.object(DataStorage, "__init__", lambda x, y: None)
+async def test_remove_room_member(
+    init_fake_room,
+    mock_dataStorage_read,
+    mock_dataStorage_update,
+    url,
+    status_code,
+    json_response,
+):
+    """
+    Leave room successfully
+    Args:
+        mock_datastorage_read (AyncMock): Asynchronous external api call
+    """
+
+    db = DataStorage("3467sd4671a5f5478df56u911")
+    db.plugin_id = "34453"
+
+    update_response = {
+        "status": 200,
+        "message": "success",
+        "data": {"matched_documents": 1, "modified_documents": 1},
+    }
+
+    mock_dataStorage_read.return_value = init_fake_room
+    mock_dataStorage_update.return_value = update_response
+
+    response = client.patch(url=url)
+
+    assert response.status_code == status_code
+    assert response.json() == json_response
+
+
+@pytest.mark.asyncio
+@mock.patch.object(DataStorage, "__init__", lambda x, y: None)
+async def test_remove_room_member_wrong_room_id(
+    mock_dataStorage_read, mock_dataStorage_update
```

Comment:
The mock_update is not needed, the code won't reach the db.update when the room is empty.